subcommand/mesh-init/command.go

```diff
@@ -127,6 +142,51 @@ func (c *Command) realRun() error {
 	return nil
 }
 
+// loginToAuthMethod runs a 'consul login' command to obtain a token.
+// The login command is skipped if LogintOptions is not set in the
+// consul-ecs config JSON, in order to support non-ACL deployments.
+func (c *Command) loginToAuthMethod(tokenFile string, taskMeta awsutil.ECSTaskMeta) error {
+	method := c.config.ConsulLogin.Method
+	if method == "" {
+		method = config.DefaultAuthMethodName
+	}
+	loginOpts := []string{
+		"login", "-type", "aws", "-method", method,
+		"-token-sink-file", tokenFile,
+		"-meta", fmt.Sprintf("consul.hashicorp.com/ecs-task-id=%s", taskMeta.TaskID()),
+		"-aws-auto-bearer-token",
+	}
+	if !c.config.ConsulLogin.NoIncludeEntity {
+		loginOpts = append(loginOpts, "-aws-include-entity")
+	}
+	if len(c.config.ConsulLogin.ExtraLoginFlags) > 0 {
+		loginOpts = append(loginOpts, c.config.ConsulLogin.ExtraLoginFlags...)
+	}
+
+	err := backoff.RetryNotify(func() error {
+		// We'll get errors until the consul binary is copied to the volume ("fork/exec: text file busy")
+		c.log.Debug("login", "cmd", fmt.Sprint(loginOpts))
+		cmd := exec.Command("consul", loginOpts...)
```

Comment:
Can we use the Consul API instead of CLI?

Reply:
No.

Well, technically, we could. We could copy over [this function](https://github.com/hashicorp/consul/blob/a1eb7744070da3a3d2a7fc4be34dafdbaddf1d88/internal/iamauth/util.go#L39) to generate the [BearerToken](https://www.consul.io/api-docs/acl#login-to-auth-method) for the /acl/login request to Consul. But I didn't think ahead to expose that in some Consul package we could import (consul/api? consul/sdk?), so the CLI does the login for now.

Reply:
Can be done later but I think it would be nice to use the API for this. Execing the consul binary can cause memory issues (we've seen it on k8s) so it's nice if we don't have to.

I think exposing that function in the api package is a good idea.